servers/src/main/java/tachyon/worker/block/BlockMasterSync.java

```diff
@@ -186,13 +191,8 @@ private void handleMasterCommand(Command cmd) throws Exception {
       // Master requests blocks to be removed from Tachyon managed space.
       case Free:
         for (long block : cmd.mData) {
-          try {
-            mBlockDataManager.removeBlock(Users.MASTER_COMMAND_USER_ID, block);
-          } catch (NotFoundException nfe) {
-            // TODO: Throw a better exception here
-            // NotFoundException will be thrown if the block is unable to be locked.
-            LOG.warn("Failed master free block cmd for: " + block + " due to concurrent read.");
-          }
+          mFixedExecutionService.execute(new BlockRemover(mBlockDataManager,
```

Comment:
Sorry for replying late, thanks for your detailed explanation! I left another comment, thanks!